### PR TITLE
✨ Table (Pricing Table) component 

### PR DESCRIPTION
**Pull request by joshbermanssw**
This addresses PBI [✨ Pricing - Tier Comparison Table](https://github.com/tinacms/tina.io/issues/2496) - took a little longer than i thought

Figma mockup: https://www.figma.com/proto/Jyj0SszjfSBc6Minkl3g1l/Tina.io?node-id=1351-9459&node-type=frame&t=Zn0HHY4EW9oELRXo-0&scaling=scale-down-width&content-scaling=fixed&page-id=673%3A2513&starting-point-node-id=1351%3A9459

**Changes:**
- [ ] Created new table component - components/blocks/table/table.tsx & table.template.tsx 
- [ ] Created my own segment of logic, similar to Isaac's in `criteriaMapping()` I have created a PBI to clean up this code and extrapolate into its own class (https://github.com/tinacms/tina.io/issues/2510)

https://github.com/user-attachments/assets/1a8e8ab2-8174-487d-86ef-cae1ca3be94a
**Figure: Scroll Behaviour on < 1200px screens**

Screenshot 2024-11-25 at 2 17 45 pm
**Figure: Laptop+ sized screen view**

### CODE REVIEW

**Comment by bettybondoc** [comment]
@joshbermanssw 
cc @Marxoz @landonmaxwell 

Looking great! A few design comments:
- [ ] Desktop - Add border-radius to the table
- [ ] Desktop - Make the pricing subheading smaller
- [ ] Desktop - Remove the first row (redundant with pricing subheading)
- [ ] Mobile - Reduce gaps between columns
- [ ] Mobile - Make the first column more compact (remove unnecessary space on the left
- [ ] Mobile - When you drag the table to the right, you get this weird space to the left of the first column - can we remove that?
- [ ] Mobile - Fix row borders on mobile

**Comment by joshbermanssw** [comment]
https://tina-ebdsw1vze-tinacms.vercel.app/pricing

**Review by isaaclombardssw — APPROVED** [review]
Very interesting to use Tina to store JSON.

Mostly LGTM, good job.

File: components/blocks/CompareBox/CompareBox.template.ts [hunk 1]
```diff
@@ -1,153 +1,185 @@
...
+            existingCell ??
+            JSON.stringify({ columnHeader: item, cellValue: '', isTicked: false })
+          );
         });
-        // Rewrite the Tina form values with the updated satisfied criteria list before saving
-        values.blocks[blockIndex].companies[index].satisfiedCriteria = updatedCriteriaSatisfaction;
-      }
-    });
-  };
+
```
[the remark this answers is not in the document]
Comment (joshbermanssw) [reply components/blocks/CompareBox/CompareBox.template.ts:65]: I made a PBI for this [♻️ Refactor - criteriaMapping()](https://github.com/tinacms/tina.io/issues/2510)

**Pull request merged**